### PR TITLE
Add support for Redis 6 `auth pass [user]`

### PR DESCRIPTION
**Pull request by gkorland**
pending https://github.com/NodeRedis/redis-commands/pull/35

### CODE REVIEW

**Comment by gkorland** [comment]
@Salakar can you please review it?

**Comment by gkorland** [comment]
@stockholmux I don't see, what am I missing?

**Comment by farzadso** [comment]
@gkorland Any updates on this?

**Comment by gkorland** [comment]
@Salakar can you please review this PR?

**Comment by farzadso** [comment]
@gkorland Looks good to me.

**Comment by gkorland** [comment]
@Salakar can you please review this PR?

**Review by stockholmux — APPROVED** [review]
LGTM

**Comment by farzadso** [comment]
Any updates on this?

**Comment by farzadso** [comment]
Will this ever be merged?

**Comment by leibale** [comment]
Not yet, it will be in the next couple if days

**Comment by leibale** [comment]
@gkorland please change the override of the `auth` function in [auth.spec.test#L64](https://github.com/NodeRedis/node-redis/blob/89945ee22073220ab51013ba70c61ead536a7f0c/test/auth.spec.js#L64) from:
```javascript
client.auth = function (pass, callback) {
  callback(null, 'retry worked');
};
```
to:
```javascript
client.auth = function (pass, user, callback) {
  callback(null, 'retry worked');
};
```

**Review by leibale — APPROVED** [review]

File: test/auth.spec.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ describe('client authentication', function () {
                 });
                 var tmp = client.command_queue.get(0).callback;
                 client.command_queue.get(0).callback = function (err, res) {
-                    client.auth = function (pass, callback) {
+                    client.auth = function (pass, user, callback) {
```
Comment (gkorland) [inline test/auth.spec.js:64]: @leibale shouldn't we keep the original test too? for backward?

Comment (leibale) [reply test/auth.spec.js:64]: @gkorland overriding a function on the client is not part of the API..

**Comment by leibale** [comment]
@gkorland IDK whats doing on, it seems like it's running old workflows

**Pull request merged**